core/src/test/java/io/grpc/internal/ManagedChannelImplTest.java

```diff
@@ -1193,6 +1193,18 @@ public void getState_loadBalancerDoesNotSupportChannelState() {
   }
 
   @Test
+  public void getState_loadBalancerDoesNotSupportChannelStateThenChannelShutdown() {
+    createChannel(new FakeNameResolverFactory(false), NO_INTERCEPTOR);
+    assertEquals(ConnectivityState.IDLE, channel.getState(false));
+    helper.updatePicker(mockPicker);
+
+    channel.shutdown();
```

Comment:
The decision for ChannelExecutor to catch instead of throwing is questionable. I did it because ChannelExecutor may run in any thread, e.g., network thread, and I didn't feel comfortable to let random code, e.g., LoadBalancer to throw in network thread. But maybe it's better to just throw? @ejona86 WDYT?